registry/interchange.go

```diff
@@ -11,18 +11,23 @@ import (
 // instances. Interchange is also used internally as part of JobGroup
 // Job type.
 type JobInterchange struct {
-	Name                 string                 `bson:"_id" json:"name" yaml:"name"`
-	Type                 string                 `json:"type" bson:"type" yaml:"type"`
-	Group                string                 `bson:"group,omitempty" json:"group,omitempty" yaml:"group,omitempty"`
-	Version              int                    `json:"version" bson:"version" yaml:"version"`
-	Priority             int                    `json:"priority" bson:"priority" yaml:"priority"`
-	Status               amboy.JobStatusInfo    `bson:"status" json:"status" yaml:"status"`
-	Scopes               []string               `bson:"scopes,omitempty" json:"scopes,omitempty" yaml:"scopes,omitempty"`
-	ApplyScopesOnEnqueue bool                   `bson:"apply_scopes_on_enqueue" json:"apply_scopes_on_enqueue,omitempty" yaml:"apply_scopes_on_enqueue,omitempty"`
-	RetryInfo            amboy.JobRetryInfo     `bson:"retry_info" json:"retry_info,omitempty" yaml:"retry_info,omitempty"`
-	TimeInfo             amboy.JobTimeInfo      `bson:"time_info" json:"time_info,omitempty" yaml:"time_info,omitempty"`
-	Job                  rawJob                 `json:"job" bson:"job" yaml:"job"`
-	Dependency           *DependencyInterchange `json:"dependency,omitempty" bson:"dependency,omitempty" yaml:"dependency,omitempty"`
+	Name          string              `bson:"_id" json:"name" yaml:"name"`
+	Type          string              `json:"type" bson:"type" yaml:"type"`
+	Group         string              `bson:"group,omitempty" json:"group,omitempty" yaml:"group,omitempty"`
+	Version       int                 `json:"version" bson:"version" yaml:"version"`
+	Priority      int                 `json:"priority" bson:"priority" yaml:"priority"`
+	Status        amboy.JobStatusInfo `bson:"status" json:"status" yaml:"status"`
+	Scopes        []string            `bson:"scopes,omitempty" json:"scopes,omitempty" yaml:"scopes,omitempty"`
+	EnqueueScopes []string            `bson:"enqueue_scopes" json:"enqueue_scopes,omitempty" yaml:"enqueue_scopes,omitempty"`
+	// TODO (EVG-15582): remove this field after all jobs containing this field
+	// have hit their 90 day TTL (2022-02-01) since no jobs will be stored with
+	// this field anymore.
+	EnqueueAllScopesDeprecated bool                   `bson:"apply_scopes_on_enqueue,omitempty" json:"apply_scopes_on_enqueue,omitempty" yaml:"apply_scopes_on_enqueue,omitempty"`
+	EnqueueAllScopes           bool                   `bson:"enqueue_all_scopes" json:"enqueue_all_scopes,omitempty" yaml:"enqueue_all_scopes,omitempty"`
```

Comment:
Should the `bson` tag have an `omitempty`?

Reply:
Yes, I changed it.